components/external-services-mock/internal/notification/handler.go

```diff
@@ -281,66 +232,28 @@ func (h *Handler) GetResponses(writer http.ResponseWriter, r *http.Request) {
 
 func (h *Handler) FailOnceResponse(writer http.ResponseWriter, r *http.Request) {
 	if h.ShouldReturnError {
-		id, ok := mux.Vars(r)["tenantId"]
-		if !ok {
-			httphelpers.WriteError(writer, errors.New("missing tenantId in url"), http.StatusBadRequest)
-			return
-		}
-
-		if _, ok = h.Mappings[id]; !ok {
-			h.Mappings[id] = make([]Response, 0, 1)
-		}
-		bodyBytes, err := ioutil.ReadAll(r.Body)
-		if err != nil {
-			httphelpers.WriteError(writer, errors.Wrap(err, "error while reading request body"), http.StatusInternalServerError)
-			return
-		}
-
-		var result interface{}
-		if err := json.Unmarshal(bodyBytes, &result); err != nil {
-			httphelpers.WriteError(writer, errors.Wrap(err, "body is not a valid JSON"), http.StatusBadRequest)
-			return
-		}
-
-		mappings := h.Mappings[id]
-		if r.Method == http.MethodPatch {
-			mappings = append(h.Mappings[id], Response{
-				Operation:   Assign,
-				RequestBody: bodyBytes,
-			})
-		}
-
-		if r.Method == http.MethodDelete {
-			applicationId, ok := mux.Vars(r)["applicationId"]
-			if !ok {
-				httphelpers.WriteError(writer, errors.New("missing applicationId in url"), http.StatusBadRequest)
-				return
+		responseFunc := func([]byte) {
+			response := struct {
+				Error string `json:"error"`
+			}{
+				Error: "failed to parse request",
 			}
-			mappings = append(h.Mappings[id], Response{
-				Operation:     Unassign,
-				ApplicationID: &applicationId,
-				RequestBody:   bodyBytes,
-			})
+			httputils.RespondWithBody(context.TODO(), writer, http.StatusBadRequest, response)
+			h.ShouldReturnError = false
 		}
 
-		h.Mappings[id] = mappings
-
-		response := struct {
-			Error string `json:"error"`
-		}{
-			Error: "failed to parse request",
-		}
-		httputils.RespondWithBody(context.TODO(), writer, http.StatusBadRequest, response)
-		h.ShouldReturnError = false
+		h.syncFAResponse(writer, r, responseFunc)
 		return
 	}
 
 	if r.Method == http.MethodPatch {
-		h.Patch(writer, r)
+		patchHandler := h.Patch
+		patchHandler(writer, r)
 	}
 
 	if r.Method == http.MethodDelete {
-		h.Delete(writer, r)
+		deleteHandler := h.Delete
+		deleteHandler(writer, r)
```

Comment:
Why this is needed? Couldn't we use just `h.Patch(writer, r)` and `h.Delete(writer, r)`

Reply:
seems the delete is not adapted?